dm/worker/server.go

```diff
@@ -289,6 +289,7 @@ func (s *Server) StartSubTask(ctx context.Context, req *pb.StartSubTaskRequest)
 	if resp.Result {
 		op1 := clientv3.OpPut(common.UpstreamSubTaskKeyAdapter.Encode(cfg.SourceID, cfg.Name), req.Task)
 		resp.Msg = s.retryWriteEctd(op1)
```

Comment:
this function return failed msg when write etcd failed? seems strange, can we refine it

Reply:
How about returning `error` directly?

Reply:
I think it's ok

Reply:
Revised in https://github.com/pingcap/dm/pull/464/commits/63859013fce5644898bab2991f9d7980020c785b.
For operations related to tasks there is no side effect. But for `OperateMysqlWorker`, when worker has been started yet but write etcd failed, `OperateMysqlWorker` will set this worker to free but actually this worker can't handle other source any more. Same problem like communication error.
This problem should be resolved by refining the design of etcd keys.